joomla/joomla-cms
# [4.0] refactor the media adapters

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/26750#issuecomment-739248621.

### Summary of Changes

This is a proposal to change the adapters from ie: `local-0` to ie: `local-images` as sated https://github.com/joomla/joomla-cms/issues/26750#issuecomment-739248621.
Probably if this is ok (I think the second part would be better if it's `md5($name)`) would be merged to #33724 for easier testing?


@joomdonation @Fedik @bembelimen your thoughts here?


### Testing Instructions
Goto: `administrator/index.php?option=com_plugins&view=plugins&filters=[type=filesystem]`
Edit `FileSystem - Local` plugin by adding a new path, eg `installation`
Check the media manager
Re-edit the previous plugin by changing the order of the folders (first installation, second images)
Check again the media manager still operates fine

Also, edit any article (or create a new one):
- with tinMCE try to upload an image with drag and drop
- try to select an image for intro or full position


### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request

There shouldn't be any difference for end-users. This PR just ensures that the adapters are not index related.

### Documentation Changes Required
